massa-execution-worker/src/context.rs

```diff
@@ -635,6 +635,54 @@ impl ExecutionContext {
         )
     }
 
+    /// Try to slash 1 roll from the denounced address.
+    ///
+    /// # Arguments
+    /// * `denounced_addr`: address to slash the roll or deferred credit from
+    pub fn try_slash_roll(
+        &mut self,
+        denounced_addr: &Address,
+        roll_count: u64,
```

Comment:
maybe let's focus on the "roll_count = 1" case

Reply:
otherwise, "try_burn_rolls" misbehaves:  if it cannot slash ALL the rolls but just a part of them, it will still completely fail and not even try to slash "what it can"

Reply:
otherwise, "try_burn_rolls" misbehaves:  if it cannot slash ALL the rolls but just a part of them, it will still completely fail and not even try to slash "what it can"

Reply:
I'll go with multi-roll slashing :)